keras_cv/models/csp_darknet.py

```diff
@@ -109,7 +112,13 @@ def CSPDarkNet(
     base_channels = int(width_multiplier * 64)
     base_depth = max(round(depth_multiplier * 3), 1)
 
-    inputs = layers.Input(shape=input_shape)
+    if input_tensor is None:
```

Comment:
looks like we can use a utility function for this.  

Let's do:
keras_cv/models/utils.py

```
def parse_model_input(input_shape, input_tensor):
    if input_tensor is None:
       ....
    # return based on these values
```